### DISCUSSION TITLE
Reloading never finishes in certain circumstances (logging a lot of data to terminal)

### DISCUSSION THREAD

**Discussion by ekeric13** [Kludex/uvicorn#1891] [Potential Issue]
There is a bug in my code and my http server is logging my body to my terminal output. This endpoint contains form data so it is logging MBs worth of binary data.
<img width="600" alt="Screenshot 2023-03-07 at 2 00 57 PM" src="https://user-images.githubusercontent.com/6489651/223563677-177c2f58-5fc5-4b38-a2ff-161d1890f9a5.png">

But this problem is compounded by the fact that is that after i hit this endpoint, when I save any changes in the file, the server never finishes reloading. It will just hang for 5+ minutes before I give up waiting and manually kill the docker container and then rebuild it.

<img width="600" alt="Screenshot 2023-03-07 at 2 06 07 PM" src="https://user-images.githubusercontent.com/6489651/223564233-a518209e-2bc1-41f8-a193-503ca661266c.png">

This is essentially the uvicorn setup I am using:
https://github.com/starlite-api/starlite-pg-redis-docker/blob/main/app/main.py#L67-L74

It seems underneath the hood uvicron uses this lib:
https://github.com/samuelcolvin/watchfiles

Please let me know if you think I need to make an issue in there.

Does the file reloading take into account the data written to one's terminal buffer?? I am really confused why I only see this behavior after testing out this one endpoint. I hit other endpoints and then I save the same exact file and it reloads just fine.

edit: Confirmed when I fixed the issue of logging to stdout the very large body the reloading worked again.

**Comment by euri10** [comment]
a minimal reproductible example would help, from the description you make I fail at seeing where it could hang

**Reply by ekeric13** [reply]
So I wasn't able to reproduce with a basic repo :(
https://github.com/ekeric13/starlite-pg-redis-docker/tree/issue-with-reloading
```
cp .env.example .env

docker-compose up

curl --request POST \
  --url http://0.0.0.0:9000/v1/authors \
  --header 'content-type: multipart/form-data' \
  --form data=@/Users/erickennedy/Documents/data.zip
```

I guess there is something more that my code is doing. The reloading is also much faster in this repo. Might try re-creating my docker image from scratch.

**Reply by ekeric13** [reply]
Okay I will try and reproduce in that repo but i think i found the key reason why i couldn't reproduce it: it is only happening when i am using vscode + debugpy to attach a remote debugger. When i run the code without a remote debugger it reloads fine.

has issues:
```
PYDEVD_DISABLE_FILE_VALIDATION=1 python -m debugpy --wait-for-client --listen 0.0.0.0:5679 -m uvicorn app.main:app --log-level $UVICORN_LOG_LEVEL --timeout-keep-alive $UVICORN_KEEPALIVE --host $UVICORN_HOST ${UVICORN_RELOAD:+--reload} --port $UVICORN_PORT
```

works fine
```
uvicorn app.main:app --log-level $UVICORN_LOG_LEVEL --timeout-keep-alive $UVICORN_KEEPALIVE --host $UVICORN_HOST ${UVICORN_RELOAD:+--reload} --port $UVICORN_PORT
```
